crawl-ref/source/actor.cc

```diff
@@ -1032,6 +1033,107 @@ void actor::collide(coord_def newpos, const actor *agent, int pow)
     }
 }
 
+/**
+ * @brief Attempts to knock this actor away from a specific tile,
+ *        using repeated one-tile knockbacks. Any LOS requirements
+ *        must be checked by the calling function.
+ * @param cause The actor responsible for the knockback.
+ * @param source The tile this actor is being knocked directly away from.
```

Comment:
This is the same as `cause->pos()` in all cases, so don't bother passing it separately.